### PR TITLE
feat: Polygon v2 updates april2024

### PR DESCRIPTION
**Pull request by Luigy-Lemon**
* @title Polygon V2 Reserve Factor Updates & Interest Rate Increases
 * @author TokenLogic & Karpatkey
 * Discussion: https://governance.aave.com/t/arfc-reserve-factor-updates-polygon-aave-v2/13937/22
 * Discussion: https://governance.aave.com/t/arfc-polygon-v2-borrow-rate-adjustments/17252'

### CODE REVIEW

File: src/20240412_AaveV2Polygon_ReserveFactorAndBorrowRateUpdates/ReserveFactorAndBorrowRateUpdates.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
+---
+title: "Polygon V2 Reserve Factor Updates & Interest Rate Increases"
+author: "@karpatkey_TokenLogic & @ChaosLabs"
+discussions: "https://governance.aave.com/t/arfc-polygon-v2-borrow-rate-adjustments/1725"
```
Comment (brotherlymite) [inline src/20240412_AaveV2Polygon_ReserveFactorAndBorrowRateUpdates/ReserveFactorAndBorrowRateUpdates.md:4]:
```suggestion
discussions: "https://governance.aave.com/t/arfc-polygon-v2-borrow-rate-adjustments/17252"
```

File: src/20240412_AaveV2Polygon_ReserveFactorAndBorrowRateUpdates/ReserveFactorAndBorrowRateUpdates.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+| ----- | -------------- |
+| DAI   | 96.00%         |
+| USDC  | 98.00%         |
+| USDT  | 97.00%         |
+
+The following parameters are to be updated as follows in the Polygon V2 Market:
+| Asset | Current Slope1 | Proposed Slope1 | Current Uoptimal | Proposed Uoptimal | Current Slope2 | Proposed Slope2 |
+| :---: | :---: | :---: | :---: | :---: | :---: | :---: |
+| BAL | 27.00% | 50.00% | 65.00% | 20.00% | 236.00% | 134.00% |
+| CRV | 17.00% | 50.00% | 25.00% | 10.00% | 392.00% | 134.00% |
```
Comment (brotherlymite) [inline src/20240412_AaveV2Polygon_ReserveFactorAndBorrowRateUpdates/ReserveFactorAndBorrowRateUpdates.md:49]:
```suggestion
| CRV | 14.00% | 50.00% | 25.00% | 10.00% | 392.00% | 134.00% |
```

File: diffs/AaveV2Polygon_ReserveFactorAndBorrowRateUpdates_20240412_before_AaveV2Polygon_ReserveFactorAndBorrowRateUpdates_20240412_after.md
Comment (brotherlymite) [inline diffs/AaveV2Polygon_ReserveFactorAndBorrowRateUpdates_20240412_before_AaveV2Polygon_ReserveFactorAndBorrowRateUpdates_20240412_after.md:1]: I would update the file, as USDC variableRateSlope1 is not up to date.

File: src/20240412_AaveV2Polygon_ReserveFactorAndBorrowRateUpdates/AaveV2Polygon_ReserveFactorAndBorrowRateUpdates_20240412.t.sol [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  function setUp() public {
+    vm.createSelectFork(vm.rpcUrl('polygon'), 55728740);
+    proposal = new AaveV2Polygon_ReserveFactorAndBorrowRateUpdates_20240412();
+  }
+
+  /**
+   * @dev executes the generic test suite including e2e and config snapshots
+   */
+  function test_defaultProposalExecution() public {
+    (ReserveConfig[] memory allConfigsBefore, ReserveConfig[] memory allConfigsAfter) = defaultTest(
```
Comment (brotherlymite) [inline src/20240412_AaveV2Polygon_ReserveFactorAndBorrowRateUpdates/AaveV2Polygon_ReserveFactorAndBorrowRateUpdates_20240412.t.sol:30]:
```suggestion
    (, ReserveConfig[] memory allConfigsAfter) = defaultTest(
```
unused

File: src/20240412_AaveV2Polygon_ReserveFactorAndBorrowRateUpdates/ReserveFactorAndBorrowRateUpdates.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+| Asset | Reserve Factor |
+| ----- | -------------- |
+| DAI   | 96.00%         |
+| USDC  | 98.00%         |
+| USDT  | 97.00%         |
+
+The following parameters are to be updated as follows in the Polygon V2 Market:
+| Asset | Current Slope1 | Proposed Slope1 | Current Uoptimal | Proposed Uoptimal | Current Slope2 | Proposed Slope2 |
+| :---: | :---: | :---: | :---: | :---: | :---: | :---: |
+| BAL | 27.00% | 50.00% | 65.00% | 20.00% | 236.00% | 134.00% |
```
Comment (brotherlymite) [inline src/20240412_AaveV2Polygon_ReserveFactorAndBorrowRateUpdates/ReserveFactorAndBorrowRateUpdates.md:48]:
```suggestion
| BAL | 22.00% | 50.00% | 65.00% | 20.00% | 236.00% | 134.00% |
```

File: lib/aave-helpers
Comment (brotherlymite) [inline lib/aave-helpers:1]: I would not update the aave-helpers submodule in this case

File: src/20240412_AaveV2Polygon_ReserveFactorAndBorrowRateUpdates/ReserveFactorAndBorrowRateUpdates.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+Over the last [8 months](https://governance-v2.aave.com/governance/proposal/284/) the Reserve Factor (RF) on Polygon v2 has gradually increased from as low as 21.00% to 99.99%. As a result, deposit yield converged to zero and the capital efficiency of Aave v2 has been reduced on Polygon.
+With the RF now approaching the maximum across all reserves, this publication seek to gradually increase the cost of capital via periodic borrow rate adjustments, every two weeks in a predictable fashion.
+Based upon recent borrow rates behaviour, the market has demonstrated the ability to withstand periods of elevated borrowing rates.
+
+## Non Frozen Assets
+
+For non-frozen assets, this proposal recommends increasing the Slope1 parameter by 75bps every two weeks, or 1.5% per month, to encourage further migration of users from v2 to v3. The outcome of this [vote](https://snapshot.org/#/aave.eth/proposal/0xe2dd228640c3cad93f5418c40c4b5743b3c6c85aa0aae9eee53cbdbca2ed5c2d) has been taken into consideration when drafting the current IR parameters included in this proposal.
```
[the remark this answers is not in the document]
Comment (Luigy-Lemon) [reply src/20240412_AaveV2Polygon_ReserveFactorAndBorrowRateUpdates/ReserveFactorAndBorrowRateUpdates.md:25]: It is increasing 75bps relative to the new adjustment to 9% across all the stable-coin reserves. I will mention that

**Review by ianflexa — APPROVED** [review]

**Comment by Luigy-Lemon** [comment]
> Foundry report
> 
> ```shell
> forge 0.2.0 (cafc260 2024-05-02T00:18:55.934780579Z)
> ```
> 
> Build log
> Test error :finnadie:❌
> ```shell
> No files changed, compilation skipped
> 
> Ran 1 test for src/20240412_AaveV2Polygon_ReserveFactorAndBorrowRateUpdates/AaveV2Polygon_ReserveFactorAndBorrowRateUpdates_20240412.t.sol:AaveV2Polygon_ReserveFactorAndBorrowRateUpdates_20240412_Test
> [FAIL. Reason: setup failed: Could not instantiate forked environment with fork url: https://polygon.llamarpc.com] setUp() (gas: 0)
> Suite result: FAILED. 0 passed; 1 failed; 0 skipped; finished in 136.42s (0.00ns CPU time)
> 
> Ran 1 test suite in 136.48s (136.42s CPU time): 0 tests passed, 1 failed, 0 skipped (1 total tests)
> 
> Failing tests:
> Encountered 1 failing test in src/20240412_AaveV2Polygon_ReserveFactorAndBorrowRateUpdates/AaveV2Polygon_ReserveFactorAndBorrowRateUpdates_20240412.t.sol:AaveV2Polygon_ReserveFactorAndBorrowRateUpdates_20240412_Test
> [FAIL. Reason: setup failed: Could not instantiate forked environment with fork url: https://polygon.llamarpc.com] setUp() (gas: 0)
> 
> Encountered a total of 1 failing tests, 0 tests succeeded
> ```

I recommend using the tenderly RPC. It's the only one i had reliability during testing on Polygon

**Comment by brotherlymite** [comment]
> > Foundry report
> > ```shell
> > forge 0.2.0 (cafc260 2024-05-02T00:18:55.934780579Z)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Build log
> > Test error :finnadie:❌
> > ```shell
> > No files changed, compilation skipped
> > 
> > Ran 1 test for src/20240412_AaveV2Polygon_ReserveFactorAndBorrowRateUpdates/AaveV2Polygon_ReserveFactorAndBorrowRateUpdates_20240412.t.sol:AaveV2Polygon_ReserveFactorAndBorrowRateUpdates_20240412_Test
> > [FAIL. Reason: setup failed: Could not instantiate forked environment with fork url: https://polygon.llamarpc.com] setUp() (gas: 0)
> > Suite result: FAILED. 0 passed; 1 failed; 0 skipped; finished in 136.42s (0.00ns CPU time)
> > 
> > Ran 1 test suite in 136.48s (136.42s CPU time): 0 tests passed, 1 failed, 0 skipped (1 total tests)
> > 
> > Failing tests:
> > Encountered 1 failing test in src/20240412_AaveV2Polygon_ReserveFactorAndBorrowRateUpdates/AaveV2Polygon_ReserveFactorAndBorrowRateUpdates_20240412.t.sol:AaveV2Polygon_ReserveFactorAndBorrowRateUpdates_20240412_Test
> > [FAIL. Reason: setup failed: Could not instantiate forked environment with fork url: https://polygon.llamarpc.com] setUp() (gas: 0)
> > 
> > Encountered a total of 1 failing tests, 0 tests succeeded
> > ```
> 
> I recommend using the tenderly RPC. It's the only one i had reliability during testing on Polygon

When merged to main, the CI uses private RPCs so we should not see these issues after being merged

**Pull request merged**